pkgs/applications/science/astronomy/kstars/default.nix

```diff
@@ -1,36 +1,86 @@
-{
-  lib, mkDerivation, extra-cmake-modules, fetchurl,
-
-  kconfig, kdoctools, kguiaddons, ki18n, kinit, kiconthemes, kio,
-  knewstuff, kplotting, kwidgetsaddons, kxmlgui, knotifyconfig,
-
-
-  qtx11extras, qtwebsockets, qtkeychain, libsecret,
-
-  eigen, zlib,
-
-  cfitsio, indi-full, xplanet, libnova, libraw, gsl, wcslib, stellarsolver
+{ lib
+, stdenv
+, mkDerivation
+, extra-cmake-modules
+, fetchFromGitHub
+, kconfig
+, kdoctools
+, kguiaddons
+, ki18n
+, kinit
+, kiconthemes
+, kio
+, knewstuff
+, kplotting
+, kwidgetsaddons
+, kxmlgui
+, knotifyconfig
+, qtx11extras
+, qtwebsockets
+, qtkeychain
+, qtdatavis3d
+, wrapQtAppsHook
+, breeze-icons
+, libsecret
+, eigen
+, zlib
+, cfitsio
+, indi-full
+, xplanet
+, libnova
+, libraw
+, gsl
+, wcslib
+, stellarsolver
+, libxisf
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "kstars";
-  version = "3.6.7";
+  version = "3.6.8";
 
-  src = fetchurl {
-    url = "mirror://kde/stable/kstars/kstars-${version}.tar.xz";
-    sha256 = "sha256-uEgzvhlHHpXyvi3Djfwg3GmYeZq+r48m7OJFIDARpe4=";
+  src = fetchFromGitHub {
+    owner = "KDE";
+    repo = "kstars";
+    rev = "b384ced";
```

Comment:
Using short revs is unsafe and we cannot do that.

Reply:
Fixed by using the `stable-3.6.9` tag rev

Reply:
That's not a tag, but a branch. I am really confused why they are doing this.
We have a similar problem with branches as in when they push to it, the hash is wrong.

Reply:
Then we can use the official download link: 
https://download.kde.org/stable/kstars/kstars-3.6.9.tar.xz ?

Reply:
That's the mirror link https://github.com/NixOS/nixpkgs/blob/8f13ee7049313e438f4f96910bedf35848e0594d/pkgs/build-support/fetchurl/mirrors.nix#L127 , so yes ofc

Reply:
> The 3.6.9 version is actually available under https://download.kde.org/stable/kstars/kstars-2.6.9.tar.xz (note the **2**.6.9 instead of **3**.6.9). Upstream ticket: https://bugs.kde.org/show_bug.cgi?id=480853

Aahhhh! ok! Thank you! So, I guess we just have to wait for that to be corrected...